public/app/core/controllers/login_ctrl.ts

```diff
@@ -19,10 +19,29 @@ export class LoginCtrl {
     $scope.disableLoginForm = config.disableLoginForm;
     $scope.disableUserSignUp = config.disableUserSignUp;
     $scope.loginHint = config.loginHint;
+    $scope.oauthAutoLogin = config.oauthAutoLogin;
 
     $scope.loginMode = true;
     $scope.submitBtnText = 'Log in';
 
+    if ($scope.oauthAutoLogin) {
+      const redirect = (url: string) => {
+        $location.path('/redirect');
+        window.location.href = url;
+      };
+      if ($scope.oauth.google) {
```

Comment:
Not sure the redirect should have handled in the frontend. Seems easier to redirect the user in the backend. Any specific reason for doing it in the frontend?

Reply:
You are absolutely right, redirecting in the backend is cleaner. I will update the PR.

Reply:
The PR is ready for review.